specification/sbi_atee.adoc

```diff
@@ -13,6 +43,858 @@ running a VMM).
+    parameters can be used to complete an emulated MMIO access. The vCPU resumes at
+    the following instruction the next time it is run.
+    */
+    mmio_page_fault = 5,
+    /* The vCPU executed a WFI instruction. */
+    wait_for_interrupt = 6,
+    /*
+    The vCPU encountered an exception that the TSM cannot handle internally and that
+    cannot be safely delegated to the host. The value of the SCAUSE register is stored
+    in ExitCause0. The vCPU is no longer runnable.
+    */
+    unhandled_exception = 7,
+    /*
+    The vCPU encountered a guest page fault in a shared memory region. The faulting 
+    guest physical address is stored in ExitCause0. The vCPU will resume at the faulting
+    instruction the next time it is run.
+    */
+    shared_page_fault = 9,
+};
+-----
+
+*Returns* the one of the values enumerated by `tvm_vcpu_run_exit_code` above in
+sbiret.value on success, and the caller should examine it to determine further
+action.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_vcpu_create_errors]
+.TEE TVM VCPU Run Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The TVM exited and sbiret.value contains the tvm_exit_code
+| SBI_ERR_INVALID_PARAM | The `tvm_guest_id` or `tvm_vcpu_id` was invalid, or guest
+                          wasn't runnable.
+|===
+
+=== Function: TEE TVM VCPU Create (FID #8)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_vcpu_create(unsigned long tvm_guest_id,
+                                      unsigned long tvm_vcpu_id);
+-----
+Adds a vCPU with ID vcpu_id to the TVM specified by `tvm_guest_id`. The `tvm_guest_id`
+is the unique identifier returned by a previous call to`sbi_tee_create_tvm()`. VCPUs may
+not be added after the TVM is finalized.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_tvm_vcpu_run_errors]
+.TEE TVM VCPU Create Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The VCPU was successfully added to the TVM
+| SBI_ERR_INVALID_PARAM | The parameters were invalid, or the TVM
+                          wasn't in "initializing" state.
+|===
+
+=== Function: TEE TVM VCPU Set Register (FID #9)
+[source, C]
+-----
+struct sbiret sbi_tee_tvm_vcpu_set_register(unsigned long tvm_guest_id,
+                                            unsigned long tvm_vcpu_id,
+                                            unsigned long tvm_vcpu_register,
+                                            unsigned long register_value);
+-----
+Sets the register identified by `tvm_vcpu_register` to the value specified by 
+`register_value` on the VCPU specified by `tvm_vcpu_id` on the TVM specified by 
+`tvm_guest_id`. The `tvm_guest_id is the unique identifier returned by a previous
```

Comment:
This is a bit of a mouthful. How about: "Sets the vCPU virtual register `tvm_vcpu_register` to `register_value`"? The meaning of `tvm_guest_id` / `tvm_vcpu_id` should be obvious from context, IMO.